numpydoc/numpydoc.py

```diff
@@ -184,6 +183,31 @@ class NumpyCDomain(ManglingDomainBase, CDomain):
     }
 
 
+def match_items(lines, content_old):
+    """Create the right items for lines.
+
+    This tries to recover where a line in ``lines``
+    came from before mangling.
+
+    It assumes that missing or new lines
+    are always empty.
```

Comment:
Do you have a better heuristic? As I said in the comment, I'm mostly concerned about knowing which method an error is in.